### PR TITLE
2.10 hmaarrfk

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by RaulPPelaez** [comment]
Only one fail now!
```shell
import: 'torch'
Traceback (most recent call last):
  File "/Users/runner/miniforge3/conda-bld/pytorch_1698510889611/test_tmp/run_test.py", line 2, in <module>
    import torch
  File "/Users/runner/miniforge3/conda-bld/pytorch_1698510889611/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.10/site-packages/torch/__init__.py", line 235, in <module>
    from torch._C import *  # noqa: F403
ImportError: dlopen(/Users/runner/miniforge3/conda-bld/pytorch_1698510889611/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.10/site-packages/torch/_C.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace (_mkl_blas_caxpy)
```

File: recipe/build.sh [hunk 1]
```diff
@@ -12,18 +12,18 @@ rm -rf pyproject.toml
...
-if [[ "$c_compiler" == "clang" ]]; then
-    export CXXFLAGS="$CXXFLAGS -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-error=unused-command-line-argument"
-    export CFLAGS="$CFLAGS -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-error=unused-command-line-argument"
-else
-    export CXXFLAGS="$CXXFLAGS -Wno-deprecated-declarations -Wno-error=maybe-uninitialized"
-    export CFLAGS="$CFLAGS -Wno-deprecated-declarations -Wno-error=maybe-uninitialized"
-fi
+# export CFLAGS="$(echo $CFLAGS | sed 's/-fvisibility-inlines-hidden//g')"
+# export CXXFLAGS="$(echo $CXXFLAGS | sed 's/-fvisibility-inlines-hidden//g')"
+# export LDFLAGS="$(echo $LDFLAGS | sed 's/-Wl,--as-needed//g')"
```
Comment (hmaarrfk) [inline recipe/build.sh:17]:
```suggestion
export LDFLAGS="$(echo $LDFLAGS | sed 's/-Wl,--as-needed//g')"
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -31,142 +27,132 @@ build:
...
+    - fsspec
+    - packaging
+    - optree
+    - libcblas              # [blas_impl != "mkl"]
+    - liblapack             # [blas_impl != "mkl"]
+    - libgomp   # [linux]
+    - llvm-openmp    # [osx]
+    - libprotobuf
+    - sleef
+    - libuv
```
Comment (hmaarrfk) [inline recipe/meta.yaml:97]: linking is saying that it isn't necessary to have these for osx-arm.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -31,142 +27,132 @@ build:
...
+    - pkg-config  # [unix]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:98]: i'm not sure if `pkg-config` is required here.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -31,142 +27,132 @@ build:
...
+    - types-dataclasses
+    - sympy
+    - filelock
+    - networkx
+    - jinja2
+    - fsspec
+    - packaging
+    - optree
+    - libcblas              # [blas_impl != "mkl"]
+    - liblapack             # [blas_impl != "mkl"]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:92]:
```suggestion
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -31,142 +27,132 @@ build:
...
+    - setuptools
+    - astunparse
+    - expecttest
+    - hypothesis
+    - numpy
+    - psutil
+    - pyyaml
+    - requests
+    - mkl-devel {{ mkl }}   # [blas_impl == "mkl"]
+    - libcblas * *_mkl      # [blas_impl == "mkl"]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:81]:
```suggestion
    - libcblas * *_mkl      # [blas_impl == "mkl"]
    - libblas               # [blas_impl != "mkl"]
    - libcblas              # [blas_impl != "mkl"]
    - liblapack             # [blas_impl != "mkl"]
```

Comment (hmaarrfk) [reply recipe/meta.yaml:81]: static analysis is showing that htis is likely needed for osx-arm

**Comment by jakirkham** [comment]
Now that PR ( https://github.com/conda-forge/pytorch-cpu-feedstock/pull/199 ) is in, should we close this?

**Comment by Tobias-Fischer** [comment]
Yes, this is all dealt with in #199

**Pull request closed without merging**